File: src/main/java/nz/co/searchwellington/repositories/FrontendContentUpdater.java
package nz.co.searchwellington.repositories;

import nz.co.searchwellington.model.Resource;
import nz.co.searchwellington.repositories.elasticsearch.ElasticSearchIndexUpdateService;

import org.apache.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class FrontendContentUpdater {
	
	private final static Logger log = Logger.getLogger(FrontendContentUpdater.class);

	private final ElasticSearchIndexUpdateService elasticSearchIndexUpdateService;
	
	@Autowired
	public FrontendContentUpdater(ElasticSearchIndexUpdateService elasticSearchIndexUpdateService) {
		this.elasticSearchIndexUpdateService = elasticSearchIndexUpdateService;
	}
	
	public void update(Resource updatedResource) {
		log.info("Updating elastic search record for resource: " + updatedResource.getName());
		elasticSearchIndexUpdateService.updateSingleContentItem(updatedResource);
	}

}

File: src/main/scala/nz/co/searchwellington/repositories/FrontendContentUpdater.scala
package nz.co.searchwellington.repositories

import nz.co.searchwellington.model.Resource
import nz.co.searchwellington.repositories.elasticsearch.ElasticSearchIndexRebuildService
import org.apache.commons.logging.LogFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import scala.concurrent.{ExecutionContext, Future}

@Component class FrontendContentUpdater @Autowired()(elasticSearchIndexRebuildService: ElasticSearchIndexRebuildService) {

  private val log = LogFactory.getLog(classOf[FrontendContentUpdater])

  def update(updatedResource: Resource)(implicit ec: ExecutionContext): Future[Boolean] = {
    log.debug("Updating elastic search record for resource: " + updatedResource.title)
    elasticSearchIndexRebuildService.index(updatedResource)
  }

}
